### PR TITLE
[Sparse] Add SparseMatrix unittest and fix docstring problem

### LINKED ISSUE #4603: Improve docstring and add test cases for sparse matrix
## 🔨Work Item
- Provide complete examples of the creation operators of sparse matrix in the docstring.
- Add pytest to check creation operators and member attributes.

**IMPORTANT:**
* This template is only for dev team to track project progress. For feature request or bug report, please use the corresponding issue templates.
* DO NOT create a new work item if the purpose is to fix an existing issue or feature request. We will directly use the issue in the project tracker.

Project tracker: https://github.com/orgs/dmlc/projects/2

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Description

Resolve #4603 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: tests/pytorch/mock_sparse/test_sparse_matrix.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import pytest
+import torch
+
+from dgl.mock_sparse import create_from_coo, create_from_csr, create_from_csc
+
+
```
Comment (isratnisa) [inline tests/pytorch/mock_sparse/test_sparse_matrix.py:6]: Let's add test for attributes too. `test_diag.py` is doing that.

**Review by isratnisa — CHANGES_REQUESTED** [review]

File: tests/pytorch/mock_sparse/test_sparse_matrix.py (hunk 1 above)
Comment (czkkkkkk) [reply tests/pytorch/mock_sparse/test_sparse_matrix.py:6]: Added test for `device`. I think it cover all the attributes now.

File: python/dgl/mock_sparse/sp_matrix.py [hunk 2]
```diff
@@ -28,24 +28,24 @@ class SparseMatrix:
...
-    3
+    >>> print(A)
+    SparseMatrix(indices=tensor([[1, 1, 2],
+        [2, 4, 3]]),
+    values=tensor([1, 1, 1]),
+    shape=(3, 5), nnz=3)
 
     Case2: Sparse matrix with row indices, col indices and values (vector).
 
     >>> ...
```
Comment (isratnisa) [inline python/dgl/mock_sparse/sp_matrix.py:39]: Remove `...`. We can provide either full example or continuation of the previous case.

File: python/dgl/mock_sparse/sp_matrix.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def __call__(self, x):
         assert len(x) == self.nnz
         return SparseMatrix(self.row, self.col, x, shape=self.shape)
 
-    def indices(self, fmt, return_shuffle=False) -> Tuple[torch.tensor, ...]:
```
Comment (isratnisa) [inline python/dgl/mock_sparse/sp_matrix.py:174]: `Torch.Tensor` instead of `torch.tensor`.

**Review by isratnisa — APPROVED** [review]
Minor comments. LGTM.

**Comment by isratnisa** [comment]
`Transpose()` functions are missing type hint for `return`.

File: python/dgl/mock_sparse/sp_matrix.py (hunk 2 above)
Comment (czkkkkkk) [reply python/dgl/mock_sparse/sp_matrix.py]: Done

File: python/dgl/mock_sparse/sp_matrix.py (hunk 3 above)
Comment (czkkkkkk) [reply python/dgl/mock_sparse/sp_matrix.py:174]: Done.

**Comment by czkkkkkk** [comment]
> `Transpose()` functions are missing type hint for `return`.

Added.

File: python/dgl/mock_sparse/sp_matrix.py [hunk 4]
```diff
@@ -24,28 +25,27 @@ class SparseMatrix:
     --------
     Case1: Sparse matrix with row indices, col indices and values (scalar).
 
-    >>> src = torch.tensor([1, 1, 2])
-    >>> dst = torch.tensor([2, 4, 3])
-    >>> val = torch.tensor([1, 1, 1])
+    >>> src = torch.Tensor([1, 1, 2])
+    >>> dst = torch.Tensor([2, 4, 3])
+    >>> val = torch.Tensor([1, 1, 1])
```
Comment (jermainewang) [inline python/dgl/mock_sparse/sp_matrix.py:30]: Should be `torch.tensor`

File: python/dgl/mock_sparse/diag_matrix.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 """dgl diagonal matrix module."""
+from __future__ import annotations
```
Comment (jermainewang) [inline python/dgl/mock_sparse/diag_matrix.py:2]: The annotations is only available in python 3.7. As DGL supports python 3.6, this perhaps may not be suitable. @BarclayII  what do you think?

File: python/dgl/mock_sparse/sp_matrix.py [hunk 6]
```diff
@@ -171,7 +171,7 @@ def __call__(self, x):
         assert len(x) == self.nnz
         return SparseMatrix(self.row, self.col, x, shape=self.shape)
 
-    def indices(self, fmt, return_shuffle=False) -> Tuple[torch.tensor, ...]:
+    def indices(self, fmt:str, return_shuffle=False) -> Tuple[torch.Tensor, ...]:
```
Comment (jermainewang) [inline python/dgl/mock_sparse/sp_matrix.py:174]: Are we preferring `fmt:str` or `fmt : str` or `fmt: str`? cc @frozenbugs

File: python/dgl/mock_sparse/sp_matrix.py [hunk 7]
```diff
@@ -251,9 +251,9 @@ def transpose(self):
         Example
         -------
 
-        >>> row = torch.tensor([1, 1, 3])
-        >>> col = torch.tensor([2, 1, 3])
-        >>> val = torch.tensor([1, 1, 2])
+        >>> row = torch.Tensor([1, 1, 3])
+        >>> col = torch.Tensor([2, 1, 3])
+        >>> val = torch.Tensor([1, 1, 2])
```
Comment (jermainewang) [inline python/dgl/mock_sparse/sp_matrix.py:256]: Should be `torch.tensor`

**Review by jermainewang — CHANGES_REQUESTED** [review]

File: python/dgl/mock_sparse/sp_matrix.py (hunk 4 above)
Comment (czkkkkkk) [reply python/dgl/mock_sparse/sp_matrix.py]: I see. Fixed.

File: python/dgl/mock_sparse/sp_matrix.py (hunk 7 above)
Comment (czkkkkkk) [reply python/dgl/mock_sparse/sp_matrix.py]: Fixed.

File: python/dgl/mock_sparse/sp_matrix.py (hunk 6 above)
Comment (frozenbugs) [reply python/dgl/mock_sparse/sp_matrix.py:174]: fmt: str
https://docs.python.org/3/library/typing.html

Comment (frozenbugs) [reply python/dgl/mock_sparse/sp_matrix.py:174]: black will take care of this.

File: python/dgl/mock_sparse/diag_matrix.py (hunk 5 above)
Comment (jermainewang) [reply python/dgl/mock_sparse/diag_matrix.py:2]: Understood. But due to the version requirement, we probably should drop this.

**Pull request merged**